docs/kb/laptop-dev-env.md

```diff
@@ -0,0 +1,224 @@
+# NovaCustom laptop development environment
+
+This document describes the process of configuring a development and debugging
+environment for NovaCustom laptops.
+
+The instruction is applicable for workstations running Fedora. If you're using
+a different distro, some commands (e.g. package installation) may be different.
+
+## Embedded Controller
+
+This section describes the development environment for EC firmware on NovaCustom
+devices. The Embedded Controller firmware is heavily based on System76 EC
+firmware and makes use of open-source debugging tools and software also
+developed by System76.
+
+### Flashing
+
+This section outlines the parts and software needed to perform external flashing
+of the Embedded Controller.
+
+| Size | 14 inch models | 16 inch models |
+| --- | --- | --- |
+| Programmer | Arduino Mega 2560 | Arduino Mega 2560 |
+| Cable | **FFC 0.5mm pitch 24 pin** | **FFC 1.0mm pitch 24 pin** |
+| Software | ecflash | ecflash |
+
+#### Preparation
+
+1. Obtain a copy of the EC firmware and tools:
+
+    ```bash
+    git clone https://github.com/Dasharo/ec.git
+    cd ec
+    git submodule update --init --checkout
+    ./scripts/deps.sh
+    ```
+
+1. Follow the on-screen instructions to finish configuring your development
+environment.
+1. Connect the Arduino to your workstation with a USB-A to USB-B cable. Do not
+connect the target laptop yet.
+1. Flash the Arduino with debugger firmware:
+
+    ```bash
+    make BOARD=arduino/mega2560
+    make BOARD=arduino/mega2560 flash
+    ```
+
+#### Flash the target laptop
+
+1. In the EC directory, build the EC flashing application:
+
+    ```bash
+    cargo build --manifest-path ecflash/Cargo.toml --example isp --release
+    ```
+
+1. Connect the Arduino to the target laptop with the appropriate FFC cable for
+   your model. If the cable is narrower than the socket on the laptop, align the
+   cable and socket to pin 0 (to the left).
+1. Connect a second cable (e.g. USB-C to USB-C) from the workstation to the
```

Comment:
Is it safe to connect VBUS with host and target?

Reply:
with usb-c it should be safe, because vbus stays at 0V until power roles are determined

Reply:
What happens if the direction cannot be determined?

Reply:
USB-PD spec says Vbus shall be vSafe0V or vSafe5V in an error condition. 5V is safe, because the PD controller will not open input or output switches until it negotiates 20V.